lib/virtio_mmio/virtio_mmio.c

```diff
@@ -29,6 +29,26 @@ extern void virtio_device_reset(struct virtio_device *vdev);
 
 void virtio_mmio_isr(struct virtio_device *vdev);
 
+#if defined(HVL_VIRTIO)
+int virtio_mmio_hvl_wait_cfg(struct virtio_device *vdev, int offset, uint32_t data);
+
+/* weak symbols */
+void virtio_mmio_shm_pool_init(void *mem, size_t size) __attribute__((weak));
+void *virtio_mmio_shm_pool_alloc(size_t size) __attribute__((weak));
+void virtio_mmio_shm_pool_free(void *ptr) __attribute__((weak));
+void virtio_mmio_hvl_ipi(void) __attribute__((weak));
+void virtio_mmio_hvl_wait(uint32_t usec) __attribute__((weak));
```

Comment:
This one already exist `metal_sleep_usec`.

Reply:
The Zephyr implementation of metal_sleep_usec uses k_sleep. Since virtio device setup happens in the context of the kernel initialization routine, k_busy_wait needs to be used, which is what virtio_mmio_hvl_wait uses in the Zephyr reference implementation of hypervisorless virtio.